src/concurrent_handshake.rs

```diff
@@ -0,0 +1,64 @@
+use super::AsyncTls;
+use pin_project_lite::pin_project;
+use std::future::Future;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+use tokio::io::{AsyncRead, AsyncWrite};
+use tokio::task::JoinHandle;
+
+/// See [`AsyncTls::concurrent_handshakes`]
+#[cfg_attr(docsrs, doc(cfg(feature = "rt")))]
+#[derive(Clone, Debug)]
+pub struct ConcurrentHandshakeTls<T>(pub(crate) T);
+impl<C, T> AsyncTls<C> for ConcurrentHandshakeTls<T>
+where
+    T: AsyncTls<C>,
+    C: AsyncRead + AsyncWrite,
+    T::AcceptFuture: Send + 'static,
+    T::Stream: Send + 'static,
+    T::Error: Send + 'static,
+{
+    type Stream = T::Stream;
+    type Error = T::Error;
+    type AcceptFuture = HandshakeJoin<T::Stream, T::Error>;
+
+    fn accept(&self, stream: C) -> Self::AcceptFuture {
+        HandshakeJoin {
+            inner: tokio::spawn(self.0.accept(stream)),
+        }
+    }
+}
+
+impl<T> ConcurrentHandshakeTls<T> {
```

Comment:
Please can we also have a `get_mut(&mut self) -> &mut T` so we don't have to reassemble the whole thing just to change the tls acceptor? 
Or otherwise maybe implement DerefMut to T ?

Reply:
how would one use that?

Reply:
Of course! sorry I was getting confused, I was thinking it was the Listener and not the acceptor which had been wrapped here

Reply:
Yep I have tested now, and this seems to be working great. I can just use:
```
listener.replace_acceptor(AsyncTls::<TcpStream>::spawning_handshakes(new_acceptor))
```